src/main/resources/META-INF/resources/primefaces/core/core.dialog.js

```diff
@@ -12,6 +12,72 @@ if (!PrimeFaces.dialog) {
+             * This assumes the root window DOM doesn't change between dialog opening and closing in a way that the selectors wouldn't match anymore which should be always the case.
+             */
+            var sourceFrames = function() {
+                var w = window;
+                var sourceFrames = [];
+                while(w.frameElement) {
+                    var parent = w.parent;
+                    if (parent.PF === undefined) {
+                        break;
+                    }
+
+                    var e = w.frameElement;
+                    var pieces = [];
+
+                    for (; e && e.tagName !== undefined; e = e.parentNode) {
+                        if (e.id && !/\s/.test(e.id)) {
+                            // If we find a parent with an id, we can use that as basis and stop there
+                            pieces.unshift(e.id);
+                            pieces.unshift('#');
+                            pieces.unshift(' > ');
+                            break;
+                        } else if (e.className) {
+                            // Without an id, we try to use a combination of :eq, class names and tag name and hope a parent has an id
+                            var classes = e.className.split(' ');
+                            var classSelectorPieces = [];
+                            for (var i in classes) {
+                                if (classes.hasOwnProperty(i) && classes[i]) {
+                                    classSelectorPieces.unshift(classes[i]);
+                                    classSelectorPieces.unshift('.');
+                                }
+                            }
+                            classSelectorPieces.unshift(e.tagName);
+
+                            var classSelector = classSelectorPieces.join('');
+                            var elems = $(e.parentNode).find(classSelector);
+                            if (elems.length > 1) {
+                                pieces.unshift(":eq(" + elems.index(e) + ")");
+                            }
+                            pieces.unshift(classSelector);
+                        } else {
+                            // Without classes, we try to work with :eq and the tag name
+                            var elems = $(e.parentNode).find(e.tagName);
+                            if (elems.length > 1) {
+                                pieces.unshift(":eq(" + elems.index(e) + ")");
+                            }
+                            pieces.unshift(e.tagName);
+                        }
+                        pieces.unshift(' > ');
+                    }
+
+                    var s = pieces.slice(1).join('');
+
+                    sourceFrames.unshift(s);
+                    w = parent;
+                };
+
+                return sourceFrames;
+            }();
+
```

Comment:
I can try to add some more comments and examples if you like, but that's just how resolving nested frames works. Not sure how else this could be implemented. It's complex because the scenario this handles is complex. I'm open for concrete suggestions, but I just don't see a simpler version of this. I already reduced to to the bare minimum IMO.

Reply:
yes please